src/journal/journald-server.c

```diff
@@ -1107,12 +1104,63 @@ void server_dispatch_message(
         dispatch_message_real(s, iovec, n, m, c, tv, priority, object_pid);
 }
 
+static int do_flush(Server *s, sd_journal *j) {
+        usec_t before_flush;
+        unsigned n = 0;
+        int r;
+
+        assert(s);
+        assert(j);
+
+        before_flush = now(CLOCK_MONOTONIC);
+
+        SD_JOURNAL_FOREACH(j) {
+                Object *o = NULL;
+                JournalFile *t, *f = j->current_file;
+
+                assert(f && f->current_offset > 0);
+
+                r = journal_file_move_to_object(f, OBJECT_ENTRY, f->current_offset, &o);
+                if (r < 0)
+                        return log_error_errno(r, "Can't read entry: %m");
+
+                t = find_journal(s, f->uid)->file;
+                assert(t);
+
+                r = journal_file_copy_entry(f, t, o, f->current_offset);
+                if (r < 0) {
```

Comment:
nit: we usually prefer code to be less indented. you could leave it like the former logic and write 
```
if (r >=0)
        continue;
```
and unindent this code block you have here.

you'd have to have `n++` twice though to be proper. not sure which is better between these two

Reply:
not sure neither, in this specific case the indentation remains still limited and make the code path dealing with errors "out" of the normal code path and therefore make it more obvious.